synapse/cortex.py

```diff
@@ -889,34 +886,25 @@ async def __anit__(self, dirn, conf=None):
 
         await self.postNexsAnit()
 
-        if self.mirror is not None:
-            await self._initCoreMirror(self.mirror)
-
-        # Fire the leadership hook once at boot
-        await self.onLeaderChange(self.nexsroot.amLeader())
+        # At this point, we're safe to make nexus changes
 
-    async def onLeaderChange(self, leader):
-        self.isleader = leader
-        # One shot to initialize storm services
-        if self.stormservices is None:
-            await self._initStormSvcs()
-        if leader:
-            return await self.startCortexLeader()
-        return await self.stopCortexLeader()
+        await self._initCoreMods()
 
-    async def startCortexLeader(self):
+    async def startAsLeader(self):
         '''
-        Indempotent actions that are done when a Cortex is a leader.
+        Run things that only a leader Cortex runs.
         '''
         if self.conf.get('cron:enable'):
             await self.agenda.start()
         await self.stormdmons.start()
+        await self._initStormSvcs()
```

Comment:
This prevents a follower from using astorm service if its only run from this callback